network/quic/net.go

```diff
@@ -0,0 +1,160 @@
+package quic
+
+import (
+	"bufio"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"io"
+	"io/ioutil"
+	"log"
+	"math/big"
+	"sync"
+	"time"
+
+	h "github.com/ConsenSys/handel"
+	"github.com/ConsenSys/handel/network"
+	quic "github.com/lucas-clemente/quic-go"
+)
+
+// Network is a handel.Network implementation using QUIC as its transport layer
+type Network struct {
+	mutex          sync.RWMutex
+	listeners      []h.Listener
+	quit           bool
+	enc            network.Encoding
+	quicListener   quic.Listener
+	sessionManager sessionManager
+}
+
+const handshakeTimeout = 2000 * time.Millisecond
+
+// NewNetwork creates Nework baked by QUIC protocol
+func NewNetwork(addr string, enc network.Encoding) (*Network, error) {
+	cfg := generateTLSConfig()
```

Comment:
this should not be a forced default - there should be a QUIC-with-insecure-TLS and QUIC-with-TLS where the config is given in parameter..